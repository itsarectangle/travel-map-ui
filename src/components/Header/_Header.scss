@import "./subcomponents/SiteLogo.scss";
@import "./subcomponents/NavLinks.scss";
@import "./subcomponents/UserHeaderContainer.scss";
@import "./subcomponents/UsernameDropdown.scss";

.header-container {
  z-index: 10;
  display: flex;
  align-items: center;
  justify-content: center;
  $height: 60px;
  background: $off-white;
  min-height: $height;
  max-height: $height;
  position: fixed;
  width: 100vw;
  .header-content {
    display: flex;
    align-items: center;
    width: 100%;
  }
  .site-logo-container {
    order: 1;
    display: flex;
    flex: 1;
    align-items: center;
    @media screen and (max-width: 575px) {
      .site-title {
        display: none;
      }
    }
  }
}

@media (min-width: 576px) {
  .site-title {
    display: flex;
    flex: 1;
    align-items: center;
    height: 100%;
    min-height: 48px;
    font-size: 24px;
    font-family: "Segoe UI", serif;
    font-weight: 100;
    padding-left: 16px;
    svg {
      height: 100%;
      width: 50%;
    }
  }
}

@media (min-width: 1200px) {
  .header-container {
    width: 100%;
  }

  .header-content {
    display: flex;
    align-items: center;
    max-width: 1560px;
  }
}

.hamburger-icon {
  width: 50px;
  height: 50px;
  background-color: #eee;
  position: relative;
  cursor: pointer;
  transition: all 0.5s ease;
  .hamburger-a {
    position: absolute;
    width: 50%;
    height: 3px;
    background-color: #000;
    top: 30%;
    left: 50%;
    transform: translate(-50%, -50%);
    z-index: 10;
    transition: all 0.25s ease-in-out;
  }

  .hamburger-b {
    position: absolute;
    width: 50%;
    height: 3px;
    background-color: #000;
    bottom: 30%;
    left: 50%;
    transform: translate(-50%, 50%);
    z-index: 10;
    transition: all 0.25s ease-in-out;
  }
}
.hamburger-icon::after {
  content: "";
  width: 50%;
  height: 3px;
  background-color: #000;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  z-index: 1;
  transition: all 0.5s ease;
}
.hamburger-icon.ham-active::after {
  width: 100%;
  height: 100%;
  background-color: #cb7678;
  position: absolute;
}
.ham-active.hamburger-a {
  background-color: rgb(248, 248, 252);
  z-index: 10;
}

.hamburger-icon.ham-active .hamburger-a {
  top: 50%;
  left: 50%;
  width: 60%;
  background-color: rgb(248, 248, 252);
  transform: translate(-50%, -50%) rotate(135deg);
}
.hamburger-icon.ham-active .hamburger-b {
  top: 50%;
  left: 50%;
  width: 60%;
  background-color: rgb(248, 248, 252);
  transform: translate(-50%, -50%) rotate(225deg);
}

.hamburger-dropdown-container {
  align-items: center;
  width: 100%;
  padding-bottom: 4px;
  background: rgb(248, 248, 252);
  z-index: 2000;
  transform: translateY(60px);
  @media (min-width: 1080px) {
    display: none;
  }
  a {
    cursor: pointer;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 16px;
    text-transform: uppercase;
    font-weight: 600;
    padding: 10px 0;
    text-decoration: none;
    color: $grey;
    letter-spacing: 1px;
    border-bottom: 4px solid $off-white;
    transition: all 0.5s ease;
    &.active {
      color: #528fb1;
      &:hover {
        background-color: rgba(158,158,158,0.2);
      }
    }
    &:hover {
      color: #528fb1;
      background-color: rgba(158,158,158,0.2);
    }
  }
}

#login-text {
  color: $primary-blue;
}
