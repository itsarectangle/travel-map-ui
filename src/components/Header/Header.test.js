import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {
  it('should render correctly in when the user is logged out', () => {
    const renderOpts = {
      userLoggedIn: false,
      avatarIndex: 1,
      color: 'green'
    }
    const component = shallow(<Header {...renderOpts} />);
  
    expect(component).toMatchSnapshot();
  });

  it('should render correctly in when the user is logged in', () => {
    const renderOpts = {
      userLoggedIn: true,
      avatarIndex: 1,
      color: 'green'
    }
    const component = shallow(<Header {...renderOpts} />);
  
    expect(component).toMatchSnapshot();
  });
});
