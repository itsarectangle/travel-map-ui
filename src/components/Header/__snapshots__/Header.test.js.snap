// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly in when the user is logged in 1`] = `
<Fragment>
  <header
    className="header-container"
  >
    <div
      className="header-content"
    >
      <div
        className="site-logo-container"
      >
        <SiteLogo />
        <NavLink
          exact={true}
          to="/"
        >
          <span
            className="site-title"
          >
            <SiteText />
          </span>
        </NavLink>
      </div>
      <div
        className="nav-menu-container"
      >
        <NavLinks
          formIsOpen={false}
          toggleFormIsOpen={[Function]}
        />
        <div
          className="nav-hamburger"
        >
          <div
            className="hamburger-icon"
            id="ham"
            onClick={[Function]}
          >
            <span
              className="hamburger-a"
            />
            <span
              className="hamburger-b"
            />
          </div>
        </div>
      </div>
      <UserHeaderContainer
        avatarIndex={1}
        color="green"
      />
    </div>
  </header>
  <div
    className="display-none"
    onClick={[Function]}
  >
    <NavLinks
      formIsOpen={false}
      handleHamburgerClick={[Function]}
      toggleFormIsOpen={[Function]}
    />
  </div>
</Fragment>
`;

exports[`Header should render correctly in when the user is logged out 1`] = `
<Fragment>
  <header
    className="header-container"
  >
    <div
      className="header-content"
    >
      <div
        className="site-logo-container"
      >
        <SiteLogo />
        <NavLink
          exact={true}
          to="/"
        >
          <span
            className="site-title"
          >
            <SiteText />
          </span>
        </NavLink>
      </div>
      <div
        className="nav-menu-container"
      >
        <NavLinks
          formIsOpen={false}
          toggleFormIsOpen={[Function]}
        />
        <div
          className="nav-hamburger"
        >
          <div
            className="hamburger-icon"
            id="ham"
            onClick={[Function]}
          >
            <span
              className="hamburger-a"
            />
            <span
              className="hamburger-b"
            />
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    className="display-none"
    onClick={[Function]}
  >
    <NavLinks
      formIsOpen={false}
      handleHamburgerClick={[Function]}
      toggleFormIsOpen={[Function]}
    />
  </div>
</Fragment>
`;
